arch/arm/core/cortex_r/irq_manage.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2013-2014 Wind River Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief ARM Cortex-R interrupt management
+ *
+ *
+ * Interrupt management: enabling/disabling and dynamic ISR
+ * connecting/replacing.  SW_ISR_TABLE_DYNAMIC has to be enabled for
+ * connecting ISRs at runtime.
+ */
+
+#include <kernel.h>
+#include <arch/cpu.h>
+#include <device.h>
+#include <irq_nextlevel.h>
+#include <sys/__assert.h>
+#include <toolchain.h>
```

Comment:
Do we need all these headers?

Reply:
Probably not all of them, this was simply copied and pasted from the `irq_manage.c` under arch/arm/core.

Reply:
Would be good to try and reduce down the set if possible.

Reply:
> Would be good to try and reduce down the set if possible.

will do, before this PR gets finalised.